grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -453,18 +454,34 @@ class ProductAvailabilityService {
         return quantityMap
     }
 
-    Map<Product, Map<Location, Integer>> getQuantityOnHandByProduct(Location[] locations) {
+    Map<Product, Map<Location, Integer>> getQuantityOnHandByProduct(Location[] locations, String[] categories, Boolean includeCategoryChildren) {
+        // Mysql throws an error when query with "category.id in (empty set)" is executed, but it works when NULL is included in this set
+        String queryWithCategories = categories ? "and category.id in (:categories)" : 'and category.id in (NULL)'
+
+        // Getting all subcategories of picked categories
+        def categoriesWithChildren = categories.collect { String categoryId ->
+            Category category = Category.load(categoryId)
+            category.getChildren() + category
```

Comment:
```suggestion
            category?.children + category
```

just in case someone somehow applies non-existing id, e.g. via URL + you don't have to call the method, because it's a transient, so you can just call `children`